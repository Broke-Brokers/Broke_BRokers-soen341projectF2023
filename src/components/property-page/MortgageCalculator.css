.mortgageCalculator_container {
    font-family: 'Arial', sans-serif;
    max-width: 600px;
    margin: 0 auto;
    background-color: #f7f7f7;
    padding: 20px;
    border-radius: 8px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  }

  .pretty-text {
    color: #EE7EA0;
    margin-bottom: 20px;
}


  /* Styling for the input fields */
  .mortgageCalculator_container input {
    width: 100%;
    padding: 10px;
    margin-bottom: 20px;
    border: 1px solid #ccc;
    border-radius: 4px;
  }
  
  /* Styling for the 'Calculate' button */
  .mortgageCalculator_container {
    /* Other styles */
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.mortgageCalculator_container button {
    background-color: #EE7EA0;
    color: white;
    padding: 10px 20px;
    border: none;
    border-radius: 4px;
    cursor: pointer;
    font-weight: bold;
    /* Remove the flex properties from the button */
}
  
  /* Hover effect for the button */
  .mortgageCalculator_container button:hover {
    background-color: #EE7EA0; /* Darker orange for hover state */
  }
  
  /* Styling for the results display */
  .mortgageCalculator_container h3 {
    background-color: white;
    color: #333;
    padding: 15px;
    margin-top: 20px;
    border-radius: 4px;
    border: 1px solid #ccc;
    text-align: center;
    font-size: 18px;
    font-weight: bold;
  }