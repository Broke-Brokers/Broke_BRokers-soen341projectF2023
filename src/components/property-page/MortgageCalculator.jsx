import React from 'react';
import '../property-page/MortgageCalculator.css';
import { useState } from 'react';


// testing

export default function MortgageCalculator() {
    const [monthlyPayment, setMonthlyPayment] = useState(0);
    const [purchasePrice, setPurchasePrice] = useState(0);
    const [downPayment, setDownPayment] = useState(0);
    const [loanTerm, setLoanTerm] = useState(0);
    const [loanInterest, setLoanInterest] = useState(0);

    const MortgageCalculate = () => {
        let principal = purchasePrice - downPayment;
        let monthlyInterest = loanInterest / 100 / 12;
        let numberOfPayments = loanTerm * 12;
        let monthlyPrice =
            (principal * [monthlyInterest * (1 + monthlyInterest) ** numberOfPayments]) /
            [(1 + monthlyInterest) ** numberOfPayments - 1];
        let roundedMonthlyPrice = Math.round(monthlyPrice);
        setMonthlyPayment(roundedMonthlyPrice);
        console.log({ principal });
    };

    return (
        <div className="mortgageCalculator_container">
            <p className = "pretty-text" > THE MORTGAGE CALCULATOR</p>
            <input
                type="number"
                placeholder="Purchase Price..."
                onChange={(event) => {
                    setPurchasePrice(parseFloat(event.target.value));
                }}
            />
            <br />
            <input
                type="number"
                placeholder="Down Payment..."
                onChange={(event) => {
                    setDownPayment(parseFloat(event.target.value));
                }}
            />
            <br />
            <input
                type="number"
                placeholder="Loan Term..."
                onChange={(event) => {
                    setLoanTerm(parseFloat(event.target.value));
                }}
            />
            <br />
            <input
                placeholder="Loan Interest ..."
                onChange={(event) => {
                    setLoanInterest(parseFloat(event.target.value));
                }}
            />
            <br />
            <button onClick={MortgageCalculate}> Calculate</button>
            <h3>Monthly Payments: $ {monthlyPayment}</h3>
        </div>
    );
}
