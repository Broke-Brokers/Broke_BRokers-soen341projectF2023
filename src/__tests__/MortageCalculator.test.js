import React from 'react';
import renderer from 'react-test-renderer';

import MortgageCalculator from '../components/property-page/MortgageCalculator';

it('performs snapshot testing', () => {
    const tree = renderer.create(<MortgageCalculator />).toJSON();
    expect(tree).toMatchSnapshot();
    });