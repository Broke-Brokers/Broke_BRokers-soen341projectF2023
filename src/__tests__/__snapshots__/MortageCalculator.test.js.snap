// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performs snapshot testing 1`] = `
<div
  className="mortgageCalculator_container"
>
  <p
    className="pretty-text"
  >
     THE MORTGAGE CALCULATOR
  </p>
  <input
    onChange={[Function]}
    placeholder="Purchase Price..."
    type="number"
  />
  <br />
  <input
    onChange={[Function]}
    placeholder="Down Payment..."
    type="number"
  />
  <br />
  <input
    onChange={[Function]}
    placeholder="Loan Term..."
    type="number"
  />
  <br />
  <input
    onChange={[Function]}
    placeholder="Loan Interest ..."
  />
  <br />
  <button
    onClick={[Function]}
  >
     Calculate
  </button>
  <h3>
    Monthly Payments: $ 
    0
  </h3>
</div>
`;
